src/Nethermind/Nethermind.BlockValidation/Handlers/ValidateBuilderSubmissionHandler.cs

```diff
@@ -0,0 +1,331 @@
+        _gasLimitCalculator = gasLimitCalculator;
+        _logger = txProcessingEnv.LogManager!.GetClassLogger();
+        _blockValidationConfig = blockValidationConfig;
+    }
+
+    private bool ValidateBlobsBundle(Transaction[] transactions, BlobsBundleV1 blobsBundle, out string? error)
+    {
+        // get sum of length of blobs of each transaction
+        int totalBlobsLength = transactions.Sum(t => t.BlobVersionedHashes!.Length);
+
+        if (totalBlobsLength != blobsBundle.Blobs.Length)
+        {
+            error = $"Total blobs length mismatch. Expected {totalBlobsLength} but got {blobsBundle.Blobs.Length}";
+            return false;
+        }
+
+        if (totalBlobsLength != blobsBundle.Commitments.Length)
+        {
+            error = $"Total commitments length mismatch. Expected {totalBlobsLength} but got {blobsBundle.Commitments.Length}";
+            return false;
+        }
+
+        if (totalBlobsLength != blobsBundle.Proofs.Length)
+        {
+            error = $"Total proofs length mismatch. Expected {totalBlobsLength} but got {blobsBundle.Proofs.Length}";
+            return false;
+        }
+
+        if (!KzgPolynomialCommitments.AreProofsValid(blobsBundle.Proofs, blobsBundle.Commitments, blobsBundle.Blobs))
+        {
+            error = "Invalid KZG proofs";
+            return false;
+        }
+
+        error = null;
+
+        _logger.Info($"Validated blobs bundle with {totalBlobsLength} blobs, commitments: {blobsBundle.Commitments.Length}, proofs: {blobsBundle.Proofs.Length}");
+
+        return true;
+    }
+
+    private BlockProcessor CreateBlockProcessor(IWorldState stateProvider)
+    {
+        return new BlockProcessor(
+            _txProcessingEnv.SpecProvider,
+            _blockValidator,
+            new Consensus.Rewards.RewardCalculator(_txProcessingEnv.SpecProvider),
+            new BlockProcessor.BlockValidationTransactionsExecutor(_txProcessingEnv.TransactionProcessor, stateProvider),
+            stateProvider,
+            _receiptStorage,
+            new BlockhashStore(_txProcessingEnv.SpecProvider, stateProvider),
+            _txProcessingEnv.LogManager,
+            new WithdrawalProcessor(stateProvider, _txProcessingEnv.LogManager!),
+            new ReceiptsRootCalculator()
+        );
+    }
+
+    private static void FinalizeStateAndBlock(IWorldState stateProvider, Block processedBlock, IReleaseSpec currentSpec, Block currentBlock, IBlockTree blockTree)
+    {
+        stateProvider.StateRoot = processedBlock.StateRoot!;
+        stateProvider.Commit(currentSpec);
+        stateProvider.CommitTree(currentBlock.Number);
+        blockTree.SuggestBlock(processedBlock, BlockTreeSuggestOptions.ForceSetAsMain);
+        blockTree.UpdateHeadBlock(processedBlock.Hash!);
+    }
+
+
+    private bool ValidatePayload(Block block, Address feeRecipient, UInt256 expectedProfit, long registerGasLimit, bool useBalanceDiffProfit, bool excludeWithdrawals, out string? error)
```

Comment:
This method is too long, it consists of few actual different steps:
1. Initial Validation
2. Building environment
3. Actual processing
4. Calculating stuff after

And should be split - into local methods.